ovn-org/ovn-kubernetes
# UDN egress integration: create logical infra for primary UDNs to egress for layer2 networks

#### What this PR does and why is it needed

This PR reacts to the creation of layer2 primary used defined networks by establishing the OVN logical infrastructure (GW routers and external switches in each node) to allow pods connected to that primary user defined network to egress the cluster.

#### Which issue(s) this PR fixes

Fixes #

#### Special notes for reviewers


#### How to verify it

Run the unit tests - this PR introduces unit tests to ensure the ovn topology created is the expected.

#### Details to documentation updates



#### Description for the changelog

Create logical infrastructure to allow egress for primary layer2 user defined networks

#### Does this PR introduce a user-facing change?

```release-note
Create logical infrastructure to allow egress for primary layer2 user defined networks
```
